processing/src/main/java/io/druid/query/topn/TopNQueryQueryToolChest.java

```diff
@@ -423,15 +430,105 @@ public Object apply(final Result<TopNResultValue> input)
   }
 
   @Override
-  public QueryRunner<Result<TopNResultValue>> preMergeQueryDecoration(QueryRunner<Result<TopNResultValue>> runner)
+  public QueryRunner<Result<TopNResultValue>> preMergeQueryDecoration(final QueryRunner<Result<TopNResultValue>> runner)
```

Comment:
Same with this method.  Can we move the meat into `mergeResults()`?  Or, if you think there's value in this being a separate method, let's give it a more semantically meaningful name for what it is doing and call into that from `mergeResults()` instead?